src/extensions/key_package_id_extension.rs

```diff
@@ -0,0 +1,39 @@
+//! # KeyPackage Identifiers
+//!
+//! > Key Package Extension
+//!
+//! 7.3. KeyPackage Identifiers
+//!
+//! Within MLS, a KeyPackage is identified by its hash (see, e.g., Section 11.2.1).
+//! The key_id extension allows applications to add an explicit,
+//! application-defined identifier to a KeyPackage.
+//!
+//! ```text
+//! opaque key_id<0..2^16-1>;
+//! ```
+//!
+
+use super::{Extension, ExtensionType};
+use crate::codec::{decode_vec, encode_vec, Cursor, VecSize};
+
+#[derive(PartialEq, Clone, Debug)]
+pub struct KeyIDExtension {
+    key_id: Vec<u8>,
+}
+
+impl KeyIDExtension {
```

Comment:
I would also expect a `new_from_keypackage` here. TODO?

Reply:
I don't think that's needed. But all this is part of the second PR (or probably the third, given how much is missing here ...)